### ISSUE TITLE
Cannot delete Neo field with nested block types

### ISSUE THREAD

**Issue by timkelty** [spicywebau/craft-neo#249]
Deleting a NEO field with _nested block types_ will throw an error. If you delete all the nested block types, then try and delete it will succeed.

```
{"error":"SQLSTATE[22003]: Numeric value out of range: 1690 BIGINT UNSIGNED value is out of range in '(`craft_local`.`structureelements`.`level` - 2)'\nThe SQL being executed was: UPDATE `structureelements` SET `level`=`level`-2, `dateUpdated`='2019-08-21 13:13:39' WHERE ((`lft` >= 10) AND (`rgt` <= 11)) AND (`root`=279)"}
```

I get the same error if I try and delete a _parent_ block type (leaving the children), and try and save the field.

Looks similar to https://github.com/spicywebau/craft-neo/issues/135 and https://github.com/spicywebau/craft-neo/issues/194, but occurring in a different spot (actions/fields/delete-field), so I figured I'd post a new issue.

Possibly related: https://github.com/craftcms/cms/issues/4474

Craft CMS 3.2.10
Neo 2.4.4

**Comment by timkelty** [comment]
Similarly, I get the same error if I try and delete a _parent_ block type (leaving the childen), and try and save.

**Comment by pvldigital** [comment]
Hi @timkelty,

Is the stack trace similar to #135?

**Comment by timkelty** [comment]
@pvldigital here's the trace: 

```
192.168.0.3 - - [26/Aug/2019:13:33:01 +0000] "POST /index.php?p=cp/actions/fields/delete-field HTTP/1.1" 200 47 "http://localhost:8089/cp/settings/fields" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36" "192.168.0.1"
192.168.0.4 -  26/Aug/2019:13:33:06 +0000 "POST /index.php" 500
2019-08-26 09:33:07 [-][1][-][error][yii\db\Exception] PDOException: SQLSTATE[22003]: Numeric value out of range: 1690 BIGINT UNSIGNED value is out of range in '(`linkedlearning_craft_local`.`structureelements`.`level` - 2)' in /app/vendor/composer/yiisoft/yii2/db/Command.php:1290
Stack trace:
#0 /app/vendor/composer/yiisoft/yii2/db/Command.php(1290): PDOStatement->execute()
#1 /app/vendor/composer/yiisoft/yii2/db/Command.php(1091): yii\db\Command->internalExecute('UPDATE `structu...')
#2 /app/vendor/composer/yiisoft/yii2/db/ActiveRecord.php(332): yii\db\Command->execute()
#3 /app/vendor/composer/creocoder/yii2-nested-sets/src/NestedSetsBehavior.php(549): yii\db\ActiveRecord::updateAll(Array, Array)
#4 /app/vendor/composer/creocoder/yii2-nested-sets/src/NestedSetsBehavior.php(477): creocoder\nestedsets\NestedSetsBehavior->moveNode(52, -2)
#5 [internal function]: creocoder\nestedsets\NestedSetsBehavior->afterUpdate(Object(yii\db\AfterSaveEvent))
#6 /app/vendor/composer/yiisoft/yii2/base/Component.php(627): call_user_func(Array, Object(yii\db\AfterSaveEvent))
#7 /app/vendor/composer/yiisoft/yii2/db/BaseActiveRecord.php(1000): yii\base\Component->trigger('afterUpdate', Object(yii\db\AfterSaveEvent))
#8 /app/vendor/composer/yiisoft/yii2/db/BaseActiveRecord.php(799): yii\db\BaseActiveRecord->afterSave(false, Array)
#9 /app/vendor/composer/yiisoft/yii2/db/ActiveRecord.php(681): yii\db\BaseActiveRecord->updateInternal(NULL)
#10 /app/vendor/composer/yiisoft/yii2/db/BaseActiveRecord.php(681): yii\db\ActiveRecord->update(true, NULL)
#11 /app/vendor/composer/creocoder/yii2-nested-sets/src/NestedSetsBehavior.php(131): yii\db\BaseActiveRecord->save(true, NULL)
#12 [internal function]: creocoder\nestedsets\NestedSetsBehavior->insertBefore(Object(craft\records\StructureElement))
#13 /app/vendor/composer/yiisoft/yii2/base/Component.php(297): call_user_func_array(Array, Array)
#14 /app/vendor/composer/craftcms/cms/src/services/Elements.php(1384): yii\base\Component->__call('insertBefore', Array)
#15 /app/vendor/composer/spicyweb/craft-neo/src/services/BlockTypes.php(497): craft\services\Elements->deleteElement(Object(benf\neo\elements\Block))
#16 /app/vendor/composer/craftcms/cms/src/services/ProjectConfig.php(865): benf\neo\services\BlockTypes->handleDeletedBlockType(Object(craft\events\ConfigEvent))
#17 [internal function]: craft\services\ProjectConfig->craft\services\{closure}(Object(craft\events\ConfigEvent))
#18 /app/vendor/composer/yiisoft/yii2/base/Component.php(627): call_user_func(Object(Closure), Object(craft\events\ConfigEvent))
#19 /app/vendor/composer/craftcms/cms/src/services/ProjectConfig.php(540): yii\base\Component->trigger('removeItem', Object(craft\events\ConfigEvent))
#20 /app/vendor/composer/craftcms/cms/src/services/ProjectConfig.php(389): craft\services\ProjectConfig->processConfigChanges('neoBlockTypes.d...', true)
#21 /app/vendor/composer/craftcms/cms/src/services/ProjectConfig.php(404): craft\services\ProjectConfig->set('neoBlockTypes.d...', NULL)
#22 /app/vendor/composer/spicyweb/craft-neo/src/services/BlockTypes.php(286): craft\services\ProjectConfig->remove('neoBlockTypes.d...')
#23 /app/vendor/composer/spicyweb/craft-neo/src/services/Fields.php(160): benf\neo\services\BlockTypes->delete(Object(benf\neo\models\BlockType))
#24 /app/vendor/composer/spicyweb/craft-neo/src/Field.php(662): benf\neo\services\Fields->delete(Object(benf\neo\Field))
#25 /app/vendor/composer/craftcms/cms/src/services/Fields.php(857): benf\neo\Field->beforeDelete()
#26 /app/vendor/composer/craftcms/cms/src/services/Fields.php(837): craft\services\Fields->deleteField(Object(benf\neo\Field))
#27 /app/vendor/composer/craftcms/cms/src/controllers/FieldsController.php(302): craft\services\Fields->deleteFieldById(9)
#28 [internal function]: craft\controllers\FieldsController->actionDeleteField()
#29 /app/vendor/composer/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#30 /app/vendor/composer/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#31 /app/vendor/composer/craftcms/cms/src/web/Controller.php(187): yii\base\Controller->runAction('delete-field', Array)
#32 /app/vendor/composer/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('delete-field', Array)
#33 /app/vendor/composer/craftcms/cms/src/web/Application.php(299): yii\base\Module->runAction('fields/delete-f...', Array)
#34 /app/vendor/composer/craftcms/cms/src/web/Application.php(566): craft\web\Application->runAction('fields/delete-f...', Array)
#35 /app/vendor/composer/craftcms/cms/src/web/Application.php(278): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#36 /app/vendor/composer/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#37 /app/public/index.php(9): yii\base\Application->run()
#38 {main}
```

**Comment by pvldigital** [comment]
Hi @timkelty, thanks for that. I've successfully replicated the issue and currently working on a fix. Will keep you updated on the progress.

**Comment by pvldigital** [comment]
@timkelty this issue should be fixed in release v2.4.5
